PHPDay 2020 (#39)
facile-it/facile-it.github.io: content/blog/eng/php-day-2020.md

---
authors: ["veronica-bolognesi"]
comments: true
date: "2020-09-10"
draft: false
share: true
categories: [English, Agile, Scrum, Kanban]

> title: "An Agile workflow"

Reviewer: Copy-paste error :smile:

Author: ```suggestion
title: "PHPDay 2020"
```

languageCode: "en-EN"
type: "post"
---

> Yesterday the 2020 edition of PHPDay was held in a remote fashion https://2020.phpday.it/talks.html.

Reviewer 2: I think it is better to write the date of the event instead of yesterday.

> On sept 9 the 2020 edition of PHPDay was held in a remote fashion https://2020.phpday.it/talks.html.

Reviewer: ```suggestion
On sept 9th the [2020 edition of PHPDay](https://2020.phpday.it/talks.html) was held in a remote fashion.
```

Having experienced the 2019 edition (indeed my first one, despite being a PHP developer from eons!) I should say that it was a little difficult to maintain the concentration for such a long event, but the effort was well worth of, given the level of both lecturers and talks.

> I would like to thanks the folks at Grups https://www.grusp.org/ for their passionate work. Try to sustain them!

Reviewer: ```suggestion
I would like to thanks the folks at [Grusp](https://www.grusp.org/) for their passionate work. Try to sustain them!
```

What follows is a mere recap of the event, with my little bit of comments. 

**Things I Wish I Knew When I Started Playing Developer by _Gary Hockin_**

A trip into developers' mindset from a ludic (and hence, engaging) viewpoint. Gary is a natural born speaker ;) and this talk deserves to be seen by anyone interested in programming, no matter their background and interests, at least once. 

**What's new in PHP 8 by _Derick Rethans_**

Personally, I did not like the style of the presentation (not the theme of slides, but how they got presented! :D) but the information provided is very important and so you should take a look here: https://derickrethans.nl/talks/php-phpday20 

**Symfony 5, the new bit by _Andreas Hucks_**

> A deep dive into Symfony 5 features and enhancements. Definitely to understand in order to evolve our software: we won't take dead ends if we know where the developers of our framework of choice are heading. Right?

Reviewer: I don't understand the second sentence here.. Can you clarify it a bit?

Author: ```suggestion
A deep dive into Symfony 5 features and enhancements. Knowing where the framework is heading will make us take better-informed decisions in our day to day work.
```

Author: @Jean85 Do you find it more understandable in this way?

Author: Concerning 

> I would suggest to follow a pattern that we used in previous articles like this one

Since we won't publish today I will take your suggestion and re-work the layout in the weekend and will also make Hugo work on my local env :)

Reviewer: Great! Seems good to me!

**Looping the Loop with SPL Iterators by _Mark Baker_**

A tour of the not-well-known iterators world and a little mea-culpa about the initially almost-not-existent SPL documentation! 🙂 
Slides: https://www.slideshare.net/MarkBakerUK/looping-the-loop-with-spl-iterators/

**Getting started with ReactPHP – Pushing real-time data to the browser by _Christian Lück_**

This is the real React! Not that other little thingy made by Facebook! An introductive talk to the react world and the world is based on: async and event-driven programming and of course promises (https://music.youtube.com/watch?v=Ok1Z65OMi1Y&feature=gws_kp_track), 

**Pragmatic Optimism: How PHP Conquered the Web by Pragmatic Optimism: How PHP Conquered the Web by _Samantha Quiñones_**

I really enjoyed the final talk that was a historic retrospective of the web development, starting from Charles Babbage up to (fast forward) CGI, the Personal Home Page Tools - PHP Tools and its evolution.
There was a point in this talk worth mentioning: sometimes people are ashamed to tell others they are using PHP because of (insert here any drawback you heard of - probably already fixed by now).

> The fact is: why we should be ashamed by declaring we are using PHP? Every problem could be sorted out by several approaches and tools, some better than others.

Reviewer: ```suggestion
The fact is: why should we be ashamed by declaring that we're using PHP? Every problem could be sorted out by several approaches and tools, some better suited than others.
```

Well, pragmatically, PHP is a tool that helps us solve our problems for a very long time and have demonstrated to listen to its community and to evolve.

> I liked the comparison with CoBOL: at the time it was published it was the perfect tool to program business applications, because the alternative were more or lest not-existent (unless you would go low-level).

Reviewer 3: ```suggestion
I liked the comparison with CoBOL: at the time it was published it was the perfect tool to program business applications, because the alternative were more or less not-existent (unless you would go low-level).
```

> I liked the comparison with CoBOL: at the time it was published it was the perfect tool to program business applications, because the alternative were more or lest not-existent (unless you would go low-level).

Reviewer 3: `more or lest not-existent` => `more or less not-existent`

So, CoBol was the right tool at the right time. And still it is... (that's because big companies find cheaper to maintain their big underdocumented monoliths than re-engineering it, but this is another story).

> So, let's be proud to be PHP developers, and let's the 2021 PHPDay edition could be held in person in Verona!

Reviewer 3: ```suggestion
So, let's be proud to be PHP developers, and let's hope that the 2021 PHPDay edition will be held in person in Verona!
```

**Pragmatic Optimism: How PHP Conquered the Web by Pragmatic Optimism: How PHP Conquered the Web by _Samantha Quiñones_**

I really enjoyed the final talk that was a historic retrospective of the web development, starting from Charles Babbage up to (fast forward) CGI, the Personal Home Page Tools - PHP Tools and its evolution.
There was a point in this talk worth mentioning: sometimes people are ashamed to tell others they are using PHP because of (insert here any drawback you heard of - probably already fixed by now).
The fact is: why we should be ashamed by declaring we are using PHP? Every problem could be sorted out by several approaches and tools, some better than others.
Well, pragmatically, PHP is a tool that helps us solve our problems for a very long time and have demonstrated to listen to its community and to evolve.
I liked the comparison with CoBOL: at the time it was published it was the perfect tool to program business applications, because the alternative were more or lest not-existent (unless you would go low-level).

> So, CoBol was the right tool at the right time. At still it is... (that's because big companies find cheaper to maintain their big underdocumented monoliths than re-engineering it, but this is another story).

Author: ```suggestion
So, CoBol was the right tool at the right time. And still it is... (that's because big companies find cheaper to maintain their big underdocumented monoliths than re-engineering it, but this is another story).
```

---
authors: ["veronica-bolognesi"]
comments: true
date: "2020-09-10"
draft: false
share: true
categories: [English, Agile, Scrum, Kanban]
title: "PHPDay 2020"

languageCode: "en-EN"
type: "post"
---

> On sept 8th the [2020 edition of PHPDay](https://2020.phpday.it/) was held in a remote fashion.

Reviewer 4: I would write:
The [2020 edition of PHPDay](https://2020.phpday.it/) took place on September 8th in remote mode.

> Having experienced the 2019 edition (indeed my first one, despite being a PHP developer from eons!) I should say that it was a little difficult to maintain the concentration for such a long event, but the effort was well worth of, given the level of both lecturers and talks.

Reviewer 4: I would change the sentence as follows:

(...) I would say that it was a bit diffcult to stay focused for such a long event. However, given the high-quality level of lectures and talks, it was totally worth the effort.

Reviewer 4: @Zolletta: ok, are you keeping the original sentence?

> I would like to thanks the folks at [Grusp](https://www.grusp.org/) for their passionate work. Try to sustain them!

Reviewer 4: to thank (without the S)

> I would like to thanks the folks at [Grusp](https://www.grusp.org/) for their passionate work. Try to sustain them!

Reviewer 4: I would write "Please support them!"

> What follows is a mere recap of the event, with my little bit of comments. 

Reviewer 4: with some comments of mine.

# Things I Wish I Knew When I Started Playing Developer
## by Gary Hockin ([@GeeH](https://twitter.com/GeeH))

> A trip into developers' mindset from a ludic (and hence, engaging) viewpoint. Gary is a natural born speaker ;) and this talk deserves to be seen by anyone interested in programming, no matter their background and interests, at least once. 

Reviewer 4: I would use "point of view" instead of viewpoint. It is way more popular.
**Hint:** otherwise, you could use "perspective".

> A trip into developers' mindset from a ludic (and hence, engaging) viewpoint. Gary is a natural born speaker ;) and this talk deserves to be seen by anyone interested in programming, no matter their background and interests, at least once. 

Reviewer 4: this talk should be seen at least once by anyone interested in programming, no matter their background and interests.

#What's new in PHP 8 
## Derick Rethans ([@derickr](https://twitter.com/derickr))

> Personally, I did not like the style of the presentation (not the theme of slides, but how they got presented! :D) but the information provided is very important and so you should take a look here: https://derickrethans.nl/talks/php-phpday20 

Reviewer 4: I would change the sentence as follows:

Personally, I did not like the style of the presentation. I'm not reffering to the graphic layout of slides but to how they were presented :D. However,  the information provided is very important so you should take a look here: https://derickrethans.nl/talks/php-phpday20

#Symfony 5, the new bit 
## Andreas Hucks ([@meandmymonkey](https://twitter.com/meandmymonkey))

> A deep dive into Symfony 5 features and enhancements. Knowing where the framework is heading will make us take better-informed decisions in our day to day work.

Reviewer 4: I would simplify the sentence as follows:

By knowing where the framework is heading, we will make better-informed decisions in our daily work.

**Note:** to make a decision is more commen than to take a decision.  (see [https://forum.wordreference.com/threads/take-a-decision-or-make-a-decision.305550/](url)

#Leveraging Typed Exceptions for Cleaner Error Handling 
## Chris Holland ([@chrisholland](https://twitter.com/chrisholland))

> A talk suited mostly to entry-level/mid-level developers, but indeed useful to learn (or recap) how to establish a lifelong relationship with exceptions: they are not enemies, they can help us contextualizing the scope in which they occur and so speed up the debugging process.

Reviewer 4: What about it?

This speech is mainly intended for entry-level or mid-level developers. However, it is very useful to learn (or recap) how to build a long-lasting relationship with exceptions. In fact, exceptions are not our enemies, they can help us contextualize the scope in which they occur. Therefore, exceptions can speed up the debugging process.

Author: Ok, but I'll change from present to past tense: this speech was

Slides available: https://bit.ly/exceptions-ftw
 
#Kubernetes for PHP developers 
## Alessandro Lai ([@AlessandroLai](https://twitter.com/AlessandroLai))

> A well-presented talk by my colleague Alessandro - Too much dashboard screenshot for me, but that's only because I am more dev than ops ;)

Reviewer 4: Do you mean " Too many screenshots of the dashboard" ?

Reviewer 4: @Zolletta I saw you are keeping it, I don't understand the exact meaning but I'm probably missing something

Author: Kept only because forgotten!

#Technically Speaking: Improve your code with documentation 
## Alexandra White ([@heyawhite](https://twitter.com/heyawhite))

> Brilliantly presented by Alexandra! A tidbit that I brought home: there are moments to be (like) Shakespeare and others not. One of those when not to be Shakespeare is when writing documentation.

Reviewer 4: Exactly!!!! ;) The same thing I said during one of our blog meetings
![image](https://user-images.githubusercontent.com/63046911/93234826-1431c480-f77d-11ea-9344-9cf6557e3462.png)

I must say that this talk lighted my interest in technical writing.

> The cherry on the cake was that the presentation had subtitles, which should definitely become a de-facto standard at least for pre-recorded videos. Very inclusive!

Reviewer 4: The icing on the cake

Author: As above, forgot to update

Reviewer 4: NP if you want to keep the original form. IMHO we should use the most common one, that's why I was suggesting this review.
Google results: 
"The icing on the cake": 22.900.000 results
"The cherry on top": 5.090.000 results
"The cherry on the cake":  2.790.000 results

> Slides: https://docs.google.com/presentation/d/1l7cjrouAGdsGae4nXyJLcGcool1l0zx5knXvL_4h04Q/edit#slide=id.g589a77319a_0_34

Reviewer 4: Me le leggo subito :)

#Looping the Loop with SPL Iterators 
## Mark Baker ([@Mark_Baker](https://twitter.com/Mark_Baker))

> A tour of the not-well-known iterators world and a little mea-culpa about the initially almost-not-existent SPL documentation! 🙂 

Reviewer 4: little-known world of iterators

> A tour of the not-well-known iterators world and a little mea-culpa about the initially almost-not-existent SPL documentation! 🙂 

Reviewer 4: What do you mean by "initially"?

Author: A tour of the not-well-known iterators world. There was also time for a little mea-culpa about the almost-not-existent documentation when the library was first published and its initial years.

Slides: https://www.slideshare.net/MarkBakerUK/looping-the-loop-with-spl-iterators/

#Getting started with ReactPHP - Pushing real-time data to the browser 
## Christian Lück ([@another_clue](https://twitter.com/another_clue))

> This is the real React! Not that other little thingy made by Facebook! An introductive talk to the react world and the world is based on: async and event-driven programming and of course promises (https://music.youtube.com/watch?v=Ok1Z65OMi1Y&feature=gws_kp_track), 

Reviewer 4: An introductory talk to the React world which is based on: async and event-driven programming and, of course, promises.

Author: An introductionary talk to the React world and the pillars which is based on:

> This is the real React! Not that other little thingy made by Facebook! An introductive talk to the react world and the world is based on: async and event-driven programming and of course promises (https://music.youtube.com/watch?v=Ok1Z65OMi1Y&feature=gws_kp_track), 

Reviewer 4: little thing

#Pragmatic Optimism: How PHP Conquered the Web
## Samantha Quiñones ([@ieatkillerbees](https://twitter.com/ieatkillerbees))

> I really enjoyed the final talk that was a historic retrospective of the web development, starting from Charles Babbage up to (fast forward) CGI, the Personal Home Page Tools - PHP Tools and its evolution.

Reviewer 4: historical retrospective on the development of the Web,

Author: it was not on the development of web, but web-depelopment (aka programming)

Author: Changed in 
I really enjoyed the final talk that was a historical retrospective of the web and web developing,

> I really enjoyed the final talk that was a historic retrospective of the web development, starting from Charles Babbage up to (fast forward) CGI, the Personal Home Page Tools - PHP Tools and its evolution.

Reviewer 4: what are you referring to with "its"?

Author: yes

There was a point in this talk worth mentioning: sometimes people are ashamed to tell others they are using PHP because of (insert here any drawback you heard of - probably already fixed by now).
The fact is: why should we be ashamed by declaring that we're using PHP? Every problem could be sorted out by several approaches and tools, some better suited than others.

> Well, pragmatically, PHP is a tool that helps us solve our problems for a very long time and have demonstrated to listen to its community and to evolve.

Reviewer 4: pragmatically speaking

> Well, pragmatically, PHP is a tool that helps us solve our problems for a very long time and have demonstrated to listen to its community and to evolve.

Reviewer 4: PHP is a tool that has helped us solve our problems for a very long time and it has proven to listen to its community and to evolve.

> I liked the comparison with CoBOL: at the time it was published it was the perfect tool to program business applications, because the alternative were more or less not-existent (unless you would go low-level).

Reviewer 4: business applications because the alternative**s** were, more or less, not-existent

> So, CoBol was the right tool at the right time. And still it is... (that's because big companies find cheaper to maintain their big underdocumented monoliths than re-engineering it, but this is another story).

Reviewer 4: that's why

**Note:** the difference between that's why and that's because is a bit subtle for us non-native speakers to understand. 
This sentence with "that's because" in italian would mean something like "CoBol era lo strumento giusto e lo è ancora perché grandi compagnie...". Quindi, cioè che fanno le grandi compagnie rende ancora CoBol il giusto strumento.
Invece tu vuoi dire che il motivo per cui le grandi compagnie preferiscono mantenere i monoliti ecc. è perché CoBol è ancora uno strumento valido.
In this case, you should use "that's why". 
Sorry for the language switch :)

Author: got it :) That's why will be!

languageCode: "en-EN"
type: "post"
---
The [2020 edition of PHPDay](https://2020.phpday.it/) took place on September 8th in a remote fashion.

Having experienced the 2019 edition (indeed my first one, despite being a PHP developer from eons!) I should say that it was a little difficult to maintain the concentration for such a long event, but the effort was well worth of, given the level of both lecturers and talks.

I would like to thank the folks at [Grusp](https://www.grusp.org/) for their passionate work. Please support them!

> What follows is a mere recap of the event, with my somme comments of mine. 

Reviewer 4: some :) Una M in più

Author: Oops

#Looping the Loop with SPL Iterators 
## Mark Baker ([@Mark_Baker](https://twitter.com/Mark_Baker))

A tour of the little-known iterators world. . There was also time for a little mea-culpa about the almost-not-existent documentation when the library was first published and its initial years.🙂 
Slides: https://www.slideshare.net/MarkBakerUK/looping-the-loop-with-spl-iterators/

#Getting started with ReactPHP - Pushing real-time data to the browser 
## Christian Lück ([@another_clue](https://twitter.com/another_clue))

> This is the real React! Not that other little thing made by Facebook! An introductory talk to the React world and the pillars is based on: async and event-driven programming and, of course, promises (https://music.youtube.com/watch?v=Ok1Z65OMi1Y&feature=gws_kp_track), 

Reviewer 4: There's a comma at the end of the sentence instead of a full stop.

> This is the real React! Not that other little thing made by Facebook! An introductory talk to the React world and the pillars is based on: async and event-driven programming and, of course, promises (https://music.youtube.com/watch?v=Ok1Z65OMi1Y&feature=gws_kp_track), 

Reviewer 4: and the pillars **it** is based on

#Pragmatic Optimism: How PHP Conquered the Web
## Samantha Quiñones ([@ieatkillerbees](https://twitter.com/ieatkillerbees))

I really enjoyed the final talk that was a historical retrospective of the web and web developing, starting from Charles Babbage up to (fast forward) CGI, the Personal Home Page Tools - PHP Tools and its evolution.
There was a point in this talk worth mentioning: sometimes people are ashamed to tell others they are using PHP because of (insert here any drawback you heard of - probably already fixed by now).
The fact is: why should we be ashamed by declaring that we're using PHP? Every problem could be sorted out by several approaches and tools, some better suited than others.
Well, pragmatically speaking, PHP is a tool that has helped us solve our problems for a very long time and it has proven to listen to its community and to evolve.

> I liked the comparison with CoBOL: at the time it was published it was the perfect tool to program business applications because the alternative were, more or less, not-existent (unless you would go low-level).

Reviewer 4: alternative**s**, S is missing
